packages/shared/src/components/auth/AuthOptions.tsx

```diff
@@ -316,7 +316,7 @@ function AuthOptions({
     onSetActiveDisplay(AuthDisplay.CodeVerification);
   };
 
-  useEventListener(globalThis, 'message', async (e) => {
+  useEventListener(broadcastChannel, 'message', async (e) => {
```

Comment:
BroadcastChannel is on the same origin yes, maybe we can merge but test with feature flag or some special query param, to not screw up production?

Reply:
Yeah, definitely we can go that route if we intend to move towards broadcast channel.

Discussed the options we have with Ido. For now, we will keep using `postMessage`, and we will only fallback to using the broadcast channel if the `window.postMessage` is `undefined`. I will just have to introduce another listener and adjust the callback page.